nupic/research/frameworks/pytorch/datasets/gsc_factory.py

```diff
@@ -0,0 +1,94 @@
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2020, Numenta, Inc.  Unless you have an agreement
+# with Numenta, Inc., for a separate license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+import os
+import tarfile
+import urllib
+
+from filelock import FileLock
+from tqdm import tqdm
+
+from nupic.research.frameworks.pytorch.dataset_utils import PreprocessedDataset
+
+DATA_URL = "http://public.numenta.com/datasets/google_speech_commands/gsc_preprocessed_v0.01.tar.gz"  # noqa: E501
+
+
+__all__ = [
+    "preprocessed_gsc",
+    "download_gsc_data",
+]
+
+
+def preprocessed_gsc(data_dir, train=True, download=True):
+    """
+    Create train or validation dataset from preprocessed GSC data, downloading if
+    necessary.
+
+    Warning: Be sure to call dataset.load_next() following each epoch of training.
+    Otherwise, no new augmentations will be loaded, and the same exact samples
+    will be reused.
+
+    :param path: path to store or load downloaded data
```

Comment:
Small typo here, the variable is called `path` in the docstring but `data_dir` above (or potentially `root` as per my comment above)